scrapy/settings/__init__.py

```diff
@@ -195,25 +193,16 @@ def getdict(self, name, default=None):
             value = json.loads(value)
         return dict(value)
 
-    def _getcomposite(self, name):
-        # DO NOT USE THIS FUNCTION IN YOUR CUSTOM PROJECTS
-        # It's for internal use in the transition away from the _BASE settings
-        # and will be removed along with _BASE support in a future release
-        basename = name + "_BASE"
-        if basename in self:
-            warnings.warn('_BASE settings are deprecated.',
-                          category=ScrapyDeprecationWarning)
-            # When users defined a _BASE setting, they explicitly don't want to
-            # use any of Scrapy's defaults. Therefore, we only use these entries
-            # from self[name] (where the defaults now live) that have a priority
-            # higher than 'default'
-            compsett = BaseSettings(self[basename], priority='default')
-            for k in self[name]:
-                prio = self[name].getpriority(k)
-                if prio > get_settings_priority('default'):
-                    compsett.set(k, self[name][k], prio)
-            return compsett
-        return self[name]
+    def getwithbase(self, name):
+        """Get a composition of a dictionary-like setting and its `_BASE`
+        counterpart.
+
+        :param name: name of the dictionary-like setting
+        :type name: string
+        """
+        compdict = self.get(name + '_BASE', {}).copy()
+        compdict.update(self.get(name, {}))
```

Comment:
Wanted to point out here that since we're promoting `<name>_BASE` settings to BaseSettings instances, `.copy()` does a deep copy of `<name>_BASE`, but we're passing `<name>` as it is when updating that copy. We could be consistent and use `self.get(name, {}).copy()` but doesn't matter too much since we don't have mutable values in default dict settings.